packages/typewiz-node/src/typewiz-node.ts

```diff
@@ -2,7 +2,14 @@
 
 import { register } from './index';
 
-register();
+(async () => {
+    if (process.argv.length > 3) {
+        await register({ typewizConfig: process.argv[2] });
```

Comment:
I think it will make more sense to search for the config file using the same heuristic ts-node uses for finding tsconfig.json, and if we want, we can also allow the user to override this using an optional command line argument (i.e. `typewiz-node --typewiz-config filename.json my-awesome-script.ts`). 

Otherwise, the current behavior can be somewhat confusing - if you run `typewiz-node my-awesome-script.ts` the script will execute, but then if you try to pass some parameter to the script, e.g. `typewiz-node my-awesome-script.ts --kittens`, it will treat the script as the configuration file, and you will get an error. 

WDYT?

Reply:
I think there are two things to consider here:
- typewiz.json resolution: As mentioned above we could put in a function which recursively searches for typewiz.json in parent directories. I think that certainly makes sense, especially for mocha integration where we can't specify a config file. [`ts-node`](https://github.com/TypeStrong/ts-node/blob/0c7fa0da66bdfd191e257becb4d07fd6e4f95c16/src/index.ts#L447) uses a [typescript function](https://github.com/Microsoft/TypeScript/blob/e6fa4e4415ececbdc8be27a1a49f620abad08179/src/compiler/program.ts#L8) for that which takes the config name as a parameter, so we should be able to use the same logic
- Argument parsing: Using a proper command line argument parser would certainly make this more robust and nicer to use. The only question for me is how do we want to handle `ts-node`? Currently you can just parse any `ts-node` argument to `typewiz-node` and those arguments will be passed along to `ts-node`. Do we want to maintain this feature (perhaps by stripping all typewiz config options from `process.argv`)?

Reply:
I see the typescript function you mentioned is exported, is it a part of the public API? can we just use it?

Regarding argument parsing, I think that stripping all typewiz config options from process.argv will do the trick. But we can skip this for now, remove the extra argument from `typewiz-node` and just stick with searching `typewiz.json` as suggested above, so we can get the PR merged sooner.

Reply:
The function is exported from the main entry point for typescript so I assume it is part of the public api.

I just added a configuration finder based on that function, it is now used by typewiz-webpack and typewiz-node. I also removed the optional argument from typewiz-node for now.